### PR TITLE
unsafe recovery: Add newly created empty regions output for unsafe recovery

### PR DESCRIPTION
**Pull request by v01dstar**
### What problem does this PR solve?

Issue Number: Ref #6715

### What is changed and how does it work?

```commit-message
In DR auto sync scenario, it is possible that a empty range hole is left in the secondary data center 
when the primary data center is totally unavailable and the replication mode is "sync recover" (super 
rare). In this case, unsafe recovery + flashback is not able to recover the cluster. The fallback 
is BR + PiTR.  While, to help the people faster identify whether the fallback is needed, we need to add 
empty regions info in the output of unsafe recovery.

If there are empty regions created, use BR+PiTR. If there are no empty regions, use flashback.
```

### Check List

Tests

- No code

Code changes

N/A

Side effects

N/A

Related changes

N/A

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: pkg/unsaferecovery/unsafe_recovery_controller.go [hunk 1]
```diff
@@ -136,6 +136,7 @@ type Controller struct {
 	// exposed to the outside for testing
 	AffectedTableIDs    map[int64]struct{}
 	affectedMetaRegions map[uint64]struct{}
+	newlyCreatedRegions map[uint64]struct{}
```
Comment (disksing) [inline pkg/unsaferecovery/unsafe_recovery_controller.go:139]: need make map in `reset()`

**Comment by v01dstar** [comment]
This PR only needs to be (auto) cherry-picked by release 7.2.

I created a manual cherry-pick PR for release 6.5 #6877

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**